src/markov/markov_approx.jl

```diff
@@ -219,3 +223,567 @@ function estimate_mc_discrete{T}(X::Vector{T})
+- `M::Intger` : number of variables of the VAR(1) model
+
+##### Returns
+
+- `A::Matirx` : impact coefficients of standardized VAR(1) process
+- `C::Matrix` : variance-covariance matrix of standardized model innovations
+- `mu::Vector` : mean of the standardized VAR(1) process
+- `Sigma::Matrix` : variance-covariance matrix of the standardized VAR(1) process
+
+"""
+function standardize_var(b::Vector, B::Matrix, Psi::Matrix, M::Integer)
+    C1 = cholfact(Psi)[:L]
+    mu = ((eye(M)-B)\eye(M))*b
+    A1 = C1\(B*C1)
+    # unconditional variance
+    Sigma1 = reshape(((eye(M^2)-kron(A1,A1))\eye(M^2))*vec(eye(M)),M,M)
+    U, _ = min_var_trace(Sigma1)
+    A = U'*A1*U
+    Sigma = U'*Sigma1*U
+    C = C1*U
+    return A, C, mu, Sigma
+end
+
+"""
+construct prior guess for evenly spaced grid method
+
+##### Arguments
+
+- `condMean::Vector` : conditional Mean of each variable
+- `Nm::Integer` : number of grid points
+- `y1D::Matrix` : grid of variable
+- `method::Even` : method for grid making
+
+"""
+construct_prior_guess(condMean::Vector, Nm::Integer, y1D::Matrix, method::Even) =
+    pdf.(Normal.(repmat(condMean, 1, Nm), 1), y1D)
+
+"""
+
+construct one-dimensional grid of states
+
+##### Argument
+
+- `Sigma::ScalarOrArray` :
+- `Nm::Integer` : number of grid points
+- `M::Integer` : number of variables (M=1 corresponds to AR(1))
+- `nSigmas::Real` : number of standard error determining end points of grid
+- `method::Even` : method for grid making
+
+##### Return
+
+- `y1D` : (M x Nm) matrix of variable grid
+
+"""
+function construct_1D_grid(Sigma::ScalarOrArray, Nm::Integer,
+                           M::Integer, nSigmas::Real, method::Even)
+    if M == 1
+        minSigmas = sqrt(Sigma)
+    else
+        minSigmas = sqrt(minimum(eig(Sigma)[1]))
+    end
+    y1Drow = collect(linspace(-minSigmas*nSigmas, minSigmas*nSigmas, Nm))'
+    y1D = repmat(y1Drow, M, 1)
+    return y1D
+end
+
+"""
+
+Return combinations of each column of matrix `A`.
+It is simiplifying `allcomb2` by using `gridmake` from QuantEcon
+
+##### Arguments
+
+- `A::Matrix` : (N x M) Matrix
+
+##### Returns
+
+- (N^M x M) Matrix, combination of each row of `A`.
+
+###### Example
+```jldoctest
+julia>allcomb3([1 4 7;
+                2 5 8;
+                3 6 9]) # numerical input
+27×3 Array{Int64,2}:
+ 1  4  7
+ 1  4  8
+ 1  4  9
+ 1  5  7
+ 1  5  8
+ 1  5  9
+ 1  6  7
+ 1  6  8
+ 1  6  9
+ 2  4  7
+ ⋮
+ 2  6  9
+ 3  4  7
+ 3  4  8
+ 3  4  9
+ 3  5  7
+ 3  5  8
+ 3  5  9
+ 3  6  7
+ 3  6  8
+ 3  6  9
+```
+
+"""
+allcomb3(A::Matrix) =
+    flipdim(gridmake(flipdim([A[:, i] for i in 1:size(A, 2)], 1)...) , 2)
+
+"""
+Compute a discrete state approximation to a distribution with known moments,
+using the maximum entropy procedure proposed in Tanaka and Toda (2013)
+
+```julia
+p, lambdaBar, momentError = discrete_approximation(D, T, TBar, q, lambda0)
+```
+
+##### Arguments
+
+- `D::Vector` : (N x 1) vector of grid points. K is the dimension of the
```

Comment:
Let's drop the ` x 1` here. I think "length `N`" fits better with how Julia arrays are represented and talked about.

Also, where does the `K` come from that you refer to on this line?